quill-core/src/main/scala/io/getquill/dsl/QueryDsl.scala

```diff
@@ -62,11 +62,19 @@ private[dsl] trait QueryDsl {
     override def filter(f: T => Boolean): EntityQuery[T]
     override def map[R](f: T => R): EntityQuery[R]
 
-    def insert: T => UnassignedAction[T] with Insert[T]
-    def insert(f: (T => (Any, Any)), f2: (T => (Any, Any))*): Insert[T]
-    def update: T => UnassignedAction[T] with Update[T]
-    def update(f: (T => (Any, Any)), f2: (T => (Any, Any))*): Update[T]
-    def delete: Delete[T]
+    def insert: T => UnassignedAction[T, Long] with Insert[T, Long]
+    def insert(f: (T => (Any, Any)), f2: (T => (Any, Any))*): Insert[T, Long]
+    def update: T => UnassignedAction[T, Long] with Update[T, Long]
+    def update(f: (T => (Any, Any)), f2: (T => (Any, Any))*): Update[T, Long]
+    def delete: Delete[T, Long]
+  }
+
+  implicit class InsertUnassignedAction[T](i: T => UnassignedAction[T, _] with Insert[T, _]) {
```

Comment:
no lucky trying to remove the implicit class. I tried to create a trait that extends `T => UnassignedAction[T] with Insert[T]`, but it breaks macros. It seems to be easier to just use implicit class.